server/api/crud/model_monitoring/deployment.py

```diff
@@ -1203,6 +1211,41 @@ def set_credentials(
             ),
         )
 
+    def _is_the_same_cred(
+        self, endpoint_store_connection, stream_path, tsdb_connection
+    ):
+        credentials_dict = {
+            key: server.api.crud.Secrets().get_project_secret(
+                project=self.project,
+                provider=mlrun.common.schemas.SecretProviderName.kubernetes,
+                secret_key=key,
+                allow_secrets_from_k8s=True,
+            )
+            for key in mlrun.common.schemas.model_monitoring.ProjectSecretKeys.mandatory_secrets()
+        }
+
+        for key, val in credentials_dict.items():
+            if (
+                key
+                == mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ENDPOINT_STORE_CONNECTION
+            ):
+                if val != endpoint_store_connection:
+                    return False
+            elif (
+                key
+                == mlrun.common.schemas.model_monitoring.ProjectSecretKeys.STREAM_PATH
+            ):
+                val = val if val is not None else "v3io"  # TODO : del in 1.9.0
+                if val != stream_path:
+                    return False
+            elif (
+                key
+                == mlrun.common.schemas.model_monitoring.ProjectSecretKeys.TSDB_CONNECTION
+            ):
+                if val != tsdb_connection:
+                    return False
```

Comment:
Can be simplified. Something like this:

```suggestion
            for key in mm_constants.ProjectSecretKeys.mandatory_secrets()
        }

        if credentials_dict[mm_constants.ProjectSecretKeys.ENDPOINT_STORE_CONNECTION] != endpoint_store_connection:
            return False
        
        val = credentials_dict[mm_constants.ProjectSecretKeys.STREAM_PATH]
        val = val if val is not None else "v3io"  # TODO: remove in 1.9.0
        if val != stream_path:
            return False
        
        if credentials_dict[mm_constants.ProjectSecretKeys.TSDB_CONNECTION] != tsdb_connection:
            return False
```

Reply:
Not sure it really simplified but ok